e2e/cypress/integration/channel/toast_spec.js

```diff
@@ -216,12 +216,21 @@ function visitTownSquareAndWaitForPageToLoad() {
 }
 
 function scrollUpAndPostAMessage() {
-    // # Scroll up so bottom is not visible
-    cy.get('div.post-list__dynamic').scrollTo(0, '70%', {duration: 1000});
+    scrollUp();
 
     // # Without the wait the tests seem to fun flaky. Possibly because of ScrollTo having a race with post of message
     cy.wait(20); // eslint-disable-line cypress/no-unnecessary-waiting
 
     // # Post a new message
     return cy.postMessageAs({sender: otherUser, message: 'This is a new message', channelId: townsquareChannelId});
 }
+
+function scrollUp() {
+    // # Scroll up so bottom is not visible
+    cy.get('div.post-list__dynamic').scrollTo(0, '70%', {duration: 1000});
+
+    // # Wait until the loading icon is not visible
+    cy.waitUntil(() => cy.get('#postListContent').within(() => {
+        return cy.get('.loading-screen').should('not.be.visible');
```

Comment:
Could you please check this again if it works as expected?  Things I observed in using `cypress-wait-until` plugin:
- use of `.within` will always return `true` so this will execute one time only and will not really wait for the element to be not visible.  I believe you should use `.then`
- `return cy.get('.loading-screen').should('not.be.visible');` I believe this will not return a value of true or false.  Should find the element returned from `.then`.

Reply:
Thanks @saturninoabril. For now, I have added a static wait after scrolling up instead of dynamically waiting, since the loader icon is actually always present and is at the very top.

Reply:
`return cy.get('.loading-screen').then(screen => Boolean(screen)));`

Reply:
Yeah, this one works.